pages/_app.js

```diff
@@ -13,14 +13,26 @@ const fetcher = async (URL) => {
 export default function App({ Component, pageProps }) {
   const [movieInfo, setMovieInfo] = useState([]);
   const [query, setQuery] = useState("");
-  const [rating ,setRating] = useState(0)
+  const [rating, setRating] = useState(0);
+  const [average, setAverage] = useState([]);
 
   const { data: movies, isLoading } = useSWR(
     `/api/movies?search=${query || "Jack+Reacher"}`,
 
     fetcher
   );
 
+  function handleAverageRating() {
+    return setAverage((average) => [...average, rating])
+     
+    
+    
+  }
+  const avrUserRating = average.reduce(
+    (acc, cur, _, arr) => acc + cur / arr.length,
+    0
+  );
```

Comment:
you need to move this code inside the Details page of your movie. Please use good variable wording, no shortcuts like "acc" , "cur" or "arr" but "average", "rating" and "array"